src/convertToObject.js

```diff
@@ -3,16 +3,34 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see
+ * an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceArray = sourceString
+    .split(';')
+    .filter(el => el.indexOf(':') !== -1);
+  const finObj = {};
+
+  sourceArray.map(callback);
+
+  function callback(element) {
```

Comment:
it's weird that you're declaring callback after it's call, and do both things inside 1 function. Either move method out of your `converToObject` func or make it a function `expression` and call it after

Reply:
@illiarerun from mozilla.org 
[Function declaration hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function#function_declaration_hoisting)
Function declarations in JavaScript are hoisted to the top of the enclosing function or global scope. You can use the function before you declared it: